src/js/ko-components/view-models/orifice-calculator.coffee

```diff
@@ -151,13 +154,40 @@ define "orifice-calculator-viewmodel", ["knockout", "lodash", "knockout.validati
           return _.ceil flowRate, 3
 
       @copyFeedbackActive = ko.observable false
-
       @copyFeedbackClass = ko.pureComputed =>
         if @copyFeedbackActive()
           return 'bounce-in'
         else
           return 'hidden'
 
+      # Fields to keep the cookies
+      @orificeBoreDiameterCookies  = ko.observableArray()
+      @baseSpecificGravityCookies  = ko.observableArray()
+      @operatingTemperatureCookies = ko.observableArray()
+      @operatingPressureCookies    = ko.observableArray()
+      @differentialPressureCookies = ko.observableArray()
+
+      @flowRate.subscribe (flowRate) =>
+        if flowRate
+          _.each FIELDS_FOR_SUGGESTION, (field) => @setCookies(field, NUMBER_OF_COOKIES)
+
+      _.each FIELDS_FOR_SUGGESTION, (field) => @initializeFieldWithCookies field
+
+    initializeFieldWithCookies: (variableName) =>
+      cookies = eval(Cookies.get(variableName))
+      if cookies
+        @["#{variableName}Cookies"] _.map cookies, (val) -> { value: val }
+
+    setCookies: (variableName, numberOfCookies) =>
+      cookies = eval(Cookies.get(variableName)) || []
```

Comment:
Lets not use eval. Instead we can rely on JSON.stringify and JSON.parse to handle these.
http://stackoverflow.com/questions/4470477/create-array-in-cookie-with-javascript

```
Don't use eval needlessly!
eval() is a dangerous function, which executes the code it's passed with the privileges of the caller. If you run eval() with a string that could be affected by a malicious party, you may end up running malicious code on the user's machine with the permissions of your webpage / extension. More importantly, third party code can see the scope in which eval() was invoked, which can lead to possible attacks in ways to which the similar Function is not susceptible.

eval() is also generally slower than the alternatives, since it has to invoke the JS interpreter, while many other constructs are optimized by modern JS engines.

There are safer (and faster!) alternatives to eval() for common use-cases.
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#Don't_use_eval_needlessly!